app/src/main/kotlin/com/wire/android/ui/home/newconversation/groupsearch/NewGroupConversationSearchPeopleScreen.kt

```diff
@@ -55,7 +55,10 @@ fun NewGroupConversationSearchPeopleScreen(
         onContactChecked = newConversationViewModel::updateSelectedContacts,
         onContinue = { navigator.navigate(NavigationCommand(NewGroupNameScreenDestination)) },
         isGroupSubmitVisible = newConversationViewModel.newGroupState.isGroupCreatingAllowed == true,
-        onClose = navigator::navigateBack,
+        onClose = {
+            newConversationViewModel.resetState()
+            navigator.navigateBack()
+        },
```

Comment:
I think this only handles the top bar navigation arrow, so we also need to add `BackHandler` to handle system back button if we want to execute some custom action before navigating back

Reply:
ohhh i missed this case, will add something